### PR TITLE
fixed README links

### PR DESCRIPTION
**Pull request by bachittle**
just a stupid small link issue I found in the readme

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -29,7 +29,7 @@ PLEASE read the zlib FAQ http://zlib.net/zlib_faq.html before asking for help.
 
 Mark Nelson <markn@ieee.org> wrote an article about zlib for the Jan.  1997
 issue of Dr.  Dobb's Journal; a copy of the article is available at
-http://marknelson.us/1997/01/01/zlib-engine/ .
+https://marknelson.us/posts/1997/01/01/zlib-engine.html.
```
Comment (fredgan) [inline README:32]: Maybe the last dot `.` should be separated by a whitespace.

**Comment by bachittle** [comment]
I also fixed all the links to use https rather than http, and a few other links that have a new redirect.

**Comment by spaette** [comment]
I prefer my approach to the Java content.

One has to click on API Documentation and then on java.base and locate java.util.zip.

At least this doesn't link against a specific JavaSE release.


```diff
- zlib is available in Java using the java.util.zip package, documented at
- http://java.sun.com/developer/technicalArticles/Programming/compression/ .
+ zlib is available in Java using the java.util.zip package, 
+ see the API Documentation for java.base at
+ https://www.oracle.com/pls/topic/lookup?ctx=en/java/javase&id=javaselatest .
```

Upon review I'd simply use https on the old CPAN address.

At least this doesn't link against a specific Perl module release.

```diff
- http://search.cpan.org/~pmqs/IO-Compress-Zlib/ .
+ https://search.cpan.org/~pmqs/IO-Compress-Zlib/ .
```

I don't know if the PR would be merged into master or simply integrated into the develop branch.

**Comment by spaette** [comment]
for reference

```
$ sed -n 15,16p IO-Compress-2.201/README
This distribution provides a Perl interface to allow reading and writing of
compressed data created with the zlib and bzip2.
$ 
```

**Comment by bachittle** [comment]
I found a closed PR in which it says that a fix is in a local branch: https://github.com/madler/zlib/pull/739, I will close the PR but leave the issue I made open.

**Comment by Neustradamus** [comment]
@madler: Have you seen this PR?

**Pull request closed without merging**